dev/com.ibm.ws.request.interrupt/test/com/ibm/ws/request/interrupt/internal/WeakReferenceThreadsTest.java

```diff
@@ -0,0 +1,131 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+
+import com.ibm.websphere.ras.TrConfigurator;
+
+import test.common.SharedOutputManager;
+
+public class WeakReferenceThreadsTest {
+
+	private static SharedOutputManager outputMgr = SharedOutputManager.getInstance();
+
+	@Rule
+	public TestRule rule = outputMgr;
+
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+		outputMgr.captureStreams();
+
+	}
+
+	@AfterClass
+	public static void tearDownAfterClass() throws Exception {
+		outputMgr.restoreStreams();
+	}
+
+	@After
+	public void tearDown() {
+		outputMgr.resetStreams();
+	}
+
+	@Test
+	/*
+	 * Tests that the weak reference for the Thread key in the WeakHashMap<Thread,
+	 * InterruptibleThreadObject> of InterruptibleThreadInfrastructureImpl is
+	 * properly removed when there are no longer any strong references to the
+	 * Thread.
+	 * 
+	 * Create an InterruptibleThreadInfrastructureImpl and initiate a new request in
+	 * a new thread. This creates a new entry in the `threadsMap` WeakHashMap with a
+	 * weak-reference to the current thread.
+	 * 
+	 * The InterruptibleThreadInfrastructureImpl would also create an
+	 * InterruptibleThreadObject which would create a JVMInterruptObject which holds
+	 * on to a reference to the current thread. This reference use to be a strong
+	 * reference and would cause a lingering reference preventing the `threadsMap`
+	 * from removing entries. For the purpose of the unit test, will create a
+	 * JVMInterruptObject separately.
+	 * 
+	 * This has been fixed to a weak-reference to the thread in JVMInterruptObject.
+	 * This test will test that once the strong reference to the running thread
+	 * (held by this test) is nullified that the remaining weak-references will
+	 * remove themselves.
+	 */
+	public void testWeakReferenceThread() throws InterruptedException {
+
+		final String REQUEST_ID = "requestID1";
+
+		/*
+		 * Hold on to reference of JVMInterruptObject outside of the separate executing
+		 * thread. Need to hold on to the JVMInterruptObject so that the object stays in
+		 * the heap and test that the (weak) Reference of the thread held by
+		 * JVMInterruptObject does not prevent the threadsMap entry in
+		 * InterruptibleThreadInfrastructureImpl from being removed.
+		 * 
+		 * Note: Previously JVMInterruptObject held a strong reference. This would fail
+		 * if there was a strong reference.
+		 * 
+		 */
+		JVMInterruptObject[] jvmInterruptRef = new JVMInterruptObject[1];
+
+		InterruptibleThreadInfrastructureImpl iti = new InterruptibleThreadInfrastructureImpl();
+
+		Runnable r = new Runnable() {
+
+			@Override
+			public void run() {
+
+				iti.newRequestEntry(REQUEST_ID);
+
+				JVMInterruptObject j = new JVMInterruptObject(null, null, null, null, null, null);
+				// Store ref to keep in heap
+				jvmInterruptRef[0] = j;
```

Comment:
I think this has no value. This problem is only in IBM Java environments. Without the code change in this PR, did this test fail?